hs_core/views/resource_folder_hierarchy.py

```diff
@@ -1033,68 +997,62 @@ def data_store_rename_file_or_folder(request, pk=None):
+        return JsonResponse({"error": "Object to be renamed does not exist"}, status=status.HTTP_400_BAD_REQUEST)
 
     if not irods_path_is_directory(istorage, src_storage_path):
         try:  # Django record should exist for each file
             ResourceFile.get(resource, base, folder=folder)
         except ResourceFile.DoesNotExist:
-            return HttpResponse('Object to be renamed does not exist',
-                                status=status.HTTP_400_BAD_REQUEST)
+            return JsonResponse({"error": "Path to be renamed does not exist"}, status=status.HTTP_400_BAD_REQUEST)
 
     # check that the target doesn't exist
     tgt_storage_path = os.path.join(resource.root_path, tgt_path)
     tgt_short_path = tgt_path[len('data/contents/'):]
     if istorage.exists(tgt_storage_path):
-        return HttpResponse('Desired name is already in use',
-                            status=status.HTTP_400_BAD_REQUEST)
+        err_msg = f"Desired name ({tgt_short_path}) already in use"
+        return JsonResponse({"error": err_msg}, status=status.HTTP_400_BAD_REQUEST)
     try:
         folder, base = ResourceFile.resource_path_is_acceptable(resource,
                                                                 tgt_storage_path,
                                                                 test_exists=False)
     except ValidationError:
-        return HttpResponse('Poorly structured desired name {}'
-                            .format(tgt_short_path),
-                            status=status.HTTP_400_BAD_REQUEST)
+        err_msg = f"Poorly structured desired name: {tgt_short_path}"
+        return JsonResponse({"error": err_msg}, status=status.HTTP_400_BAD_REQUEST)
+
     try:
         ResourceFile.get(resource, base, folder=tgt_short_path)
-        return HttpResponse('Desired name {} is already in use'
-                            .format(tgt_short_path),
-                            status=status.HTTP_400_BAD_REQUEST)
+        err_msg = f"Desired name ({tgt_short_path}) already in use"
+        return JsonResponse({"error": err_msg}, status=status.HTTP_400_BAD_REQUEST)
+
     except ResourceFile.DoesNotExist:
         pass  # correct response
 
     try:
         rename_file_or_folder(user, pk, src_path, tgt_path)
     except SessionException as ex:
-        return HttpResponse(ex.stderr, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
-    except DRF_ValidationError as ex:
-        return HttpResponse(ex.detail, status=status.HTTP_400_BAD_REQUEST)
+        return JsonResponse({"error": ex.stderr}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+    except (DRF_ValidationError, SuspiciousFileOperation) as ex:
+        err_msg = ex.detail if isinstance(ex, DRF_ValidationError) else str(ex)
+        return JsonResponse({"error": err_msg}, status=status.HTTP_400_BAD_REQUEST)
 
-    return_object = {'target_rel_path': tgt_path}
+    return JsonResponse({'target_rel_path': tgt_path})
```

Comment:
I like all of the JsonResponse modifications!
Do we feel confident that there aren't other services relying on these endpoints?
I'm thinking especially the "public" ones
https://github.com/hydroshare/hydroshare/blob/2538-filename-restrictions/hs_rest_api/urls.py#L155

One example in hsclient
https://github.com/hydroshare/hsclient/blob/master/hsclient/hydroshare.py#L565

Reply:
I will double check on this. But I am confident it should not break the public api. This is due to the following equivalence:

```
data = {"key": "value"}
HttpResponse(json.dumps(data), content_type='application/json')  = JsonResponse(data)
```

Reply:
works for me! thanks